### PR TITLE
Pg service + sslmode support 

### PR DESCRIPTION
**Pull request by domi4484**
# Pg service:
Peek 2022-03-01 10-58

# sslmode:
Peek 2022-03-04 16-42-sslmode

### CODE REVIEW

File: QgisModelBaker/gui/workflow_wizard/project_creation_page.py [hunk 1]
```diff
@@ -74,10 +74,22 @@ def _create_project(self):
             generator.stdout.connect(self.workflow_wizard.log_panel.print_info)
             generator.new_message.connect(self.workflow_wizard.log_panel.show_message)
             self.progress_bar.setValue(30)
-        except (DBConnectorError, FileNotFoundError):
+        except DBConnectorError as dB_connector_error:
```
Comment (signedav) [inline QgisModelBaker/gui/workflow_wizard/project_creation_page.py:77]:
```suggestion
        except DBConnectorError as db_connector_error:
```

File: QgisModelBaker/gui/workflow_wizard/project_creation_page.py [hunk 2]
```diff
@@ -74,10 +74,22 @@ def _create_project(self):
...
             self.workflow_wizard.log_panel.txtStdout.setText(
                 self.tr(
-                    "There was an error connecting to the database. Check connection parameters."
+                    "There was an error connecting to the database. Check connection parameters. Error details: {0}".format(
+                        dB_connector_error
```
Comment (signedav) [inline QgisModelBaker/gui/workflow_wizard/project_creation_page.py:81]:
```suggestion
                        db_connector_error
```

File: QgisModelBaker/gui/panel/pg_config_panel.py [hunk 3]
```diff
@@ -152,3 +280,112 @@ def is_valid(self):
             result = True
 
         return result, message
+
+    def _pg_service_combo_box_changed(self):
+
+        service = self.pg_service_combo_box.currentData()
```
Comment (signedav) [inline QgisModelBaker/gui/panel/pg_config_panel.py:286]: maybe return when `service == self._current_service`?

File: QgisModelBaker/utils/db_utils.py [hunk 4]
```diff
@@ -114,7 +114,21 @@ def get_db_connector(configuration):
 
     try:
         return db_factory.get_db_connector(uri_string, schema)
-    except (DBConnectorError, FileNotFoundError):
+    except DBConnectorError as dB_connector_error:
```
Comment (signedav) [inline QgisModelBaker/utils/db_utils.py:117]:
```suggestion
    except DBConnectorError as db_connector_error:
```

File: QgisModelBaker/utils/db_utils.py [hunk 5]
```diff
@@ -114,7 +114,21 @@ def get_db_connector(configuration):
...
+        QgsMessageLog.logMessage(
+            "There was an error connecting to the database. Check connection parameters. Error details: {0}".format(
+                dB_connector_error
```
Comment (signedav) [inline QgisModelBaker/utils/db_utils.py:120]:
```suggestion
                db_connector_error
```

**Review by signedav — APPROVED** [review]
Looks good. Do you think tests would make sense or is it a huge effort because it needs to "simulate" a pgservice.conf etc?

File: QgisModelBaker/gui/generate_project.py [hunk 6]
```diff
@@ -405,14 +405,25 @@ def accepted(self, edited_command=None):
                 generator.stdout.connect(self.print_info)
                 generator.new_message.connect(self.show_message)
                 self.progress_bar.setValue(30)
-            except (DBConnectorError, FileNotFoundError):
-                self.txtStdout.setText(
+            except DBConnectorError as dB_connector_error:
+                self.workflow_wizard.log_panel.txtStdout.setText(
```
[the remark this answers is not in the document]
Comment (domi4484) [reply QgisModelBaker/gui/generate_project.py:409]: ups to much copy paste without looking!

File: QgisModelBaker/gui/panel/pg_config_panel.py (hunk 3 above)
Comment (domi4484) [reply QgisModelBaker/gui/panel/pg_config_panel.py:286]: the first time running `self.pg_service_combo_box` is `None` and maybe also the `currentData()` is `None`. In that case the underlying code should still be executed

**Comment by signedav** [comment]
autoflake is not happy with our try-except import :thinking:

**Comment by domi4484** [comment]
> autoflake is not happy with our try-except import thinking

Strange with the try-except packaging import he his happy. Maybe because it in an "from ... import"

**Comment by domi4484** [comment]
@signedav should `./scripts/package_pip_packages.sh` be called somewhere in the test environment?

**Comment by signedav** [comment]
I think you are right.

**Comment by signedav** [comment]
```
      - name: Package PyPI Packages
        run: |
          sudo ./scripts/package_pip_packages.sh
```
here https://github.com/opengisch/QgisModelBaker/blob/master/.github/workflows/continuous_integration.yml#L28

**Comment by signedav** [comment]
I can imagine that autoflake would be neither happy with packaging but it only checks the files that are changed.

File: .docker/run-docker-tests.sh [hunk 7]
```diff
@@ -32,9 +32,6 @@ do
 done
 echo ""
 
-# Install needed packages
-/usr/src/scripts/package_pip_packages.sh
-
 pushd /usr/src
```
Comment (signedav) [inline .docker/run-docker-tests.sh:35]: Oh haven't remembered that they are installed here already. Obviously with problems. Do you know why?

Anyway the solution we have now is more proper I guess.

**Review by signedav — APPROVED** [review]
Okay, it's fine I guess. 

Let me think about autoflake, but merge it for now.

File: .docker/run-docker-tests.sh (hunk 7 above)
Comment (domi4484) [reply .docker/run-docker-tests.sh:35]: No this one was just a test from me. I think was not working as the path was wrong in the container

**Pull request merged**